src/gui/guiFormSpecMenu.cpp

```diff
@@ -4098,28 +4235,27 @@ enum ButtonEventType : u8
 
 bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 {
-	if (event.EventType==EET_KEY_INPUT_EVENT) {
+	if (event.EventType == EET_KEY_INPUT_EVENT) {
 		KeyPress kp(event.KeyInput);
-		if (event.KeyInput.PressedDown && (
-				(kp == EscapeKey) || (kp == CancelKey) ||
-				((m_client != NULL) && (kp == getKeySetting("keymap_inventory"))))) {
-			tryClose();
-			return true;
-		}
 
-		if (m_client != NULL && event.KeyInput.PressedDown &&
-				(kp == getKeySetting("keymap_screenshot"))) {
-			m_client->makeScreenshot();
+		if (m_use_key_event) {
+			current_keys_pending.pending_key_event = true;
+			current_keys_pending.key_event = event.KeyInput;
```

Comment:
from my understanding, it looks like this only allows on key event to be queued at a time. It's possible for the OS to deliver more than one key press when the program polls for input. This is especially likely if the program's game loop has just frozen for a period

Reply:
Oh yes.  I dealt with this problem in #10524, but not in this PR.  Will fix.

Reply:
Actually, now that I'm actually looking at this again, this isn't a problem.  When `GUIFormSpecMenu` receives an event in `OnEvent()`, if it's a key or mouse event, it dumps it into `current_keys_pending` and ships it off to `acceptInput()`, which then builds the `fields` table that you get in `minetest.register_on_player_receive_fields()` and sends it to the server.  Hence, it's a direct line from `OnEvent()` to the server, and `OnEvent()` can get called as many times a frame as Irrlicht so pleases.